lib/improver/metadata/forecast_times.py

```diff
@@ -146,6 +109,70 @@ def forecast_period_coord(
     return result_coord
 
 
+def _calculate_forecast_period(time_coord, frt_coord, fp_units,
```

Comment:
I noticed this new function doesn't have any tests of its own. I would think that some of the tests for forecast_period_coord could be moved down a level to become tests for this function instead, so we could simplify the tests for the forecast_period_coord function. What do you think?

Reply:
I think you're right.  But could I make a follow-on PR to do this, as it doesn't interact with the CLI work and I'd like not to be rushing this so I do it properly.